common/src/main/java/bisq/common/taskrunner/Task.java

```diff
@@ -65,7 +65,7 @@ protected void failed(String message) {
 
     protected void failed(Throwable t) {
         log.error(errorMessage, t);
-        taskHandler.handleErrorMessage(errorMessage);
+        taskHandler.handleErrorMessage(t.getMessage());
```

Comment:
I would like to exclude this change from this PR. It would affect the UI notifications that depend on the errormessage being reasonable for display.

Generally refactoring is fine (although more care is needed now with the many big PRs coming in, to avoid conflicts) and behavioral changes are fine as long as the ramifications are taken into account, I feel that's lacking here.

That's the only issue I have left. The rest looks good.

Reply:
I put `Task.java` back to the way it was before I made any changes to it. It now sends `An error occurred at task: ValidateOffer` up to the CLI, instead of the details of the error, i.e., `Amount is larger than 1.00 BTC`, but solving this is for another PR, with better analysis of how a solution affects the UI.